### PR TITLE
[Feature][Engine Starter] update SeaTunnel Starter and change -e to -m

### LINKED ISSUE #3674: [Feature][Engine Starter] update SeaTunnel Starter and change -e to -m
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

In starter `-e` is to specify driver run mode and only spark use, it have two select options `client`,`cluster`. And `-m` is to specify the engine deploy mode.  

SeaTunnel Engine now use `-e` to specify the engine deploy mode, we need update it and use `-m` to specify the engine deploy mode.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by FWLamb**
## Purpose of this pull request

close https://github.com/apache/incubator-seatunnel/issues/3674

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Review by chaorongzhi — DISMISSED** [review]

**Comment by chaorongzhi** [comment]
Sorry, I accidentally made a mistake.
image

**Comment by TyrantLucifer** [comment]
> Sorry, I accidentally made a mistake. image

It's no matter, thank you for your review.

**Review by hailin0 — DISMISSED** [review]
LGTM

**Comment by Hisoka-X** [comment]
@FWLamb Hi, Please fix ci problem. Thanks
image

**Comment by FWLamb** [comment]
> @FWLamb Hi, Please fix ci problem. Thanks image

I'll fix it later. Thanks!

File: seatunnel-core/seatunnel-starter/src/main/java/org/apache/seatunnel/core/starter/seatunnel/args/ClientCommandArgs.java [hunk 1]
```diff
@@ -49,6 +49,14 @@ public class ClientCommandArgs extends AbstractCommandArgs {
         description = "Get job status by JobId")
     private String jobId;
 
+    @Parameter(names = {"-can", "--cancel-job"},
+        description = "Cancel job by JobId")
+    private String cancelJobId;
+
+    @Parameter(names = {"-m", "--metrics"},
```
Comment (FWLamb) [inline seatunnel-core/seatunnel-starter/src/main/java/org/apache/seatunnel/core/starter/seatunnel/args/ClientCommandArgs.java]: @Hisoka-X @EricJoy2048 Hello, The latest code `-m` parameter has been used. Does this PR still need to be done? What should I do?

Comment (Hisoka-X) [reply seatunnel-core/seatunnel-starter/src/main/java/org/apache/seatunnel/core/starter/seatunnel/args/ClientCommandArgs.java]: how about use `-d`? mean deploy mode.

**Pull request closed without merging**